### PR TITLE
Update Flex API

### PR DESCRIPTION
**Pull request by emplums**
The upgrade to styled-system v5 meant that all flex properties got extended to both `Flex` and `Flex.Item`. It doesn't make a ton of sense to have two different props categories that refer to the same thing, so I've removed `FLEX_CONTAINER` and `FLEX_ITEM` in favor of `FLEX`.

I'm keeping `Flex.Item` around for backwards compatibility, but removing the docs for it to encourage folks to use `Flex` for everything now 🎉 This API is much better anyways because there can be instances where you want something to have both parent flex and flex item props on the same element.

### CODE REVIEW

File: pages/components/docs/Flex.md [hunk 1]
```diff
@@ -1,41 +1,36 @@
 # Flex and Flex.Item
```
Comment (colebemis) [inline pages/components/docs/Flex.md]: Should we remove `Flex.Item` from this heading?
```suggestion
# Flex
```

File: pages/components/docs/Flex.md [hunk 2]
```diff
@@ -1,41 +1,36 @@
...
 
-Flex and Flex.Item are wrapping components that will give the content flexbox properties. Flex.Item is included in the export for Flex.
+Flex is a component that will allow you to use flex box properties
+
+*** Previously, `Flex.Item` was used for flex item specific properties - we've added all properties to the `Flex` component, but are keeping Flex.Item around for backwards compatibility. ***
```
Comment (colebemis) [inline pages/components/docs/Flex.md]: Does `***` mean anything to the markdown parser?

File: pages/components/docs/Flex.md [hunk 3]
```diff
@@ -1,41 +1,36 @@
 # Flex and Flex.Item
 
-Flex and Flex.Item are wrapping components that will give the content flexbox properties. Flex.Item is included in the export for Flex.
+Flex is a component that will allow you to use flex box properties
```
Comment (colebemis) [inline pages/components/docs/Flex.md]: "Flexbox" should probably be one word. Also, do we want to call them `properties` or `system props`?

```suggestion
Flex is a component that will allow you to use flexbox properties.
```

File: pages/components/docs/Flex.md (hunk 1 above)
Comment (emplums) [reply pages/components/docs/Flex.md]: oh yis i forgot to push up my commit that did this lol

File: pages/components/docs/Flex.md (hunk 2 above)
Comment (emplums) [reply pages/components/docs/Flex.md]: Yeah I changed it to just one star *for italic text*

**Pull request merged**